### DISCUSSION TITLE
Adapt Template based on active file's contents: adjust header level

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [SilentVoid13/Templater#966] [Help]
Hello, 

I have a rudimendary logging template that follows this schematic:

    # Log [[<% tp.date.now("DD.MM.YYYY") %>]]
    whatever I am logging.
    whatever I am logging.
    whatever I am logging.

However, depending on the file the log section is at a varying header depth, e.g in one file it's a first-level header, in the next a third-level. Some files have more than one logging section.

Is it possible to detect the closest-above heading containing the word "log", retrieve its header level (number of `#`) of that header; then determine that the template's header should be one level deeper? How does one do so if the active file has mutliple logging sections?

Any thoughts on how this could be done would be appreciated.

Thank you,
Sincerely,
~Gw

**Comment by AB1908** [comment]
Given the type of questions you've been asking, I'm assuming that you can do some JS by now. You actually need to get header positions using an Obsidian API: `app.metadataCache.getFileCache(tfile)`. This gives you back an object that should have some header information that you can then write JS on to figure out where you should be inserting stuff.
